connectors/flink/src/main/scala/io/delta/flink/internal/KernelSnapshotDelegator.scala

```diff
@@ -62,6 +65,18 @@ class KernelSnapshotDelegator(
     standaloneDeltaLog: DeltaLogImpl)
   extends SnapshotImpl(hadoopConf, path, -1, LogSegment.empty(path), -1, standaloneDeltaLog, -1) {
 
+  // For backward compatibility
+  def this(
```

Comment:
this is an internal API; we don't need to provide backward compatibility guarantees. Change the existing code (if any remain) to use the new constructor.

Reply:
I think it will trigger failed binary compatibility check in our tests without this guarantee?

Reply:
@scottsand-db Do you know why we are doing Mima check for the internal classes?

Reply:
@EstherBear, you can add a line for this class [here](https://github.com/delta-io/delta/blob/master/project/FlinkMimaExcludes.scala#L27). Thank you!